AlfrescoDocumentPickerFileProvider/FileProvider.m

```diff
@@ -492,4 +478,230 @@ - (nullable NSFileProviderItemIdentifier)persistentIdentifierForItemAtURL:(NSURL
+                                                               atURL:(NSURL *)url
+{
+    NSURL *destinationURL = [self.fileService localFilesURLForFilename:[AFPItemIdentifier filenameFromItemIdentifier:itemIdentifier]];
+    
+    __weak typeof(self) weakSelf = self;
+    [self.fileCoordinator coordinateReadingItemAtURL:url
+                                             options:NSFileCoordinatorReadingForUploading
+                                    writingItemAtURL:destinationURL
+                                             options:NSFileCoordinatorWritingForReplacing
+                                               error:nil
+                                          byAccessor:^(NSURL *newReadingURL, NSURL *newWritingURL)
+    {
+        __strong typeof(self) strongSelf = weakSelf;
+        [strongSelf.fileService saveDocumentAtURL:newReadingURL
+                                            toURL:newWritingURL
+                          overwritingExistingFile:YES];
+    }];
+}
+
+- (void)handleItemChangeActionForSyncDocumentWithItemIdentifier:(NSFileProviderItemIdentifier)itemIdentifier
+                                                          atURL:(NSURL *)url {
+    NSString *accountIdentifier = [AFPItemIdentifier getAccountIdentifierFromEnumeratedIdentifier:itemIdentifier];
+    NSString *docSyncIdentifier = [AFPItemIdentifier alfrescoIdentifierFromItemIdentifier:itemIdentifier];
+    RealmSyncNodeInfo *syncNode = [[AFPDataManager sharedManager] syncItemForId:docSyncIdentifier
+                                                           forAccountIdentifier:accountIdentifier];
+    AlfrescoDocument *alfrescoDoc = (AlfrescoDocument *)syncNode.alfrescoNode;
+    NSURL *destinationURL = [NSURL fileURLWithPath:[[[RealmSyncCore sharedSyncCore] syncContentDirectoryPathForAccountWithId:accountIdentifier] stringByAppendingPathComponent:syncNode.syncContentPath]];
+    
+    __weak typeof(self) weakSelf = self;
+    [self prepareFileUploadForEditedItemAtURL:url
+                               destinationURL:destinationURL
+                          withCompletionBlock:^(NSURL *readingURL, NSError *error)
+     {
+         __strong typeof(self) strongSelf = weakSelf;
+         
+         [strongSelf uploadEditedItem:alfrescoDoc
+                   withItemIdentifier:itemIdentifier
+                                atURL:readingURL
+                 forAccountIdentifier:accountIdentifier];
+     }];
+}
+
+- (void)handleItemChangeActionForNewSyncDocumentWithItemIdentifier:(NSFileProviderItemIdentifier)itemIdentifier
+                                                             atURL:(NSURL *)url
+{
+    NSString *accountIdentifier = [AFPItemIdentifier getAccountIdentifierFromEnumeratedIdentifier:itemIdentifier];
+    AFPItemMetadata *itemMetadata = [[AFPDataManager sharedManager] metadataItemForIdentifier:itemIdentifier];
+    
+    if (itemMetadata.alfrescoNode) {
+        RLMRealm *realm = [[AFPDataManager sharedManager] realm];
```

Comment:
The realm should be obtained from RealmSyncCore based on accountIdentifier since below the object is used to save sync nodes.